src/hotspot/share/services/threadService.cpp

```diff
@@ -695,7 +695,7 @@ void ThreadStackTrace::dump_stack_at_safepoint(int maxDepth, ObjectMonitorsHasht
                         RegisterMap::UpdateMap::include,
                         RegisterMap::ProcessFrames::include,
                         RegisterMap::WalkContinuation::skip);
-
+    ResourceMark rm;
```

Comment:
Nit: Use `rm(VMThread::vm_thread());` to avoid the need to call `Thread::current()`.

Reply:
FWIW, I don't see the appeal to micro optimize away one call to Thread::current() in a function that performs a massive amount of work.

Reply:
In runtime, we prefer to always use an available reference to the current thread than look it up via `Thread::current()`.